### ISSUE TITLE
[Question] What are the correct steps to follow to fine-tune a self pre-trained BERT model on a text classification task?

### ISSUE THREAD

**Issue by FerchichiNourchene** [NVIDIA/NeMo#1267]
**Question Description**

I have pre-trained BERT model on a low resourced African dialect data with my own datasets, using the [bert_pretraining.py](https://github.com/NVIDIA/NeMo/blob/main/examples/nlp/language_modeling/bert_pretraining.py), which gave good results. Now that I have my `PretrainingBERTFromText--end.ckpt`, I would like to fine-tune my model on a text classification task based on [text_classification_with_bert.py](https://github.com/NVIDIA/NeMo/blob/main/examples/nlp/text_classification/text_classification_with_bert.py). 

I have pre-processed my own African dialect data to have the correct format `[WORD][SPACE][WORD][SPACE][WORD][TAB][LABEL] `. I have changed the` model.language_model.lm_checkpoint` in the [config file](https://github.com/NVIDIA/NeMo/blob/main/examples/nlp/text_classification/conf/text_classification_config.yaml) to my pre-trained model checkpoint path `./PretrainingBERTFromText--end.ckpt`

I had no bugs or error. But, I can't make sure if I did it the right way or if there are other steps/params in the config file to be added or changed: How to make sure, for example, that the trainer is using my African dialect LM model, and that it is not automatically using any predefined English bert-base-uncased LM model?
Should I add anything related to the `model.tokenizer`, as the language is totally different from English?

**Environment Overview**

 - Environment location: GCP
 - Method of NeMo install: pip install nemo_toolkit[all]==1.0.0b1 (latest version)

**Environment Details**

- pytorch=1.6.0
- python=3.7.9

Any suggestions from your part would really help. Thank you very much!

**Comment by okuchaiev** [comment]
@ekmb and @VahidooX  could you please comment on this?

**Comment by ericharper** [comment]
`text_classification_with_bert.py` will use the checkpoint file as long as you specify it in the config (or from the command line).

The code is here: https://github.com/NVIDIA/NeMo/blob/b90a75fcbcb792aded754bd96a2082381c3e7d48/nemo/collections/nlp/modules/common/lm_utils.py#L91

The tokenizer for text classification should be the same tokenizer that you used for pre-training. If you used the default tokenizer for each script, then you should be okay.
pre-training: https://github.com/NVIDIA/NeMo/blob/b90a75fcbcb792aded754bd96a2082381c3e7d48/examples/nlp/language_modeling/conf/bert_pretraining_from_text_config.yaml#L49
text classification: https://github.com/NVIDIA/NeMo/blob/b90a75fcbcb792aded754bd96a2082381c3e7d48/examples/nlp/text_classification/conf/text_classification_config.yaml#L38

To make sure that you're model is performing well, look at the validation metrics that we log. You should see the metrics improving on the validation set as you train for more epochs.

**Comment by FerchichiNourchene** [comment]
@ericharper Thank you very much for the detailed explanation. This actually make sens for me!
I will go for the fine-tuning step following your recommendations.
Thanks again
